### PR TITLE
init basic components

### PR DESCRIPTION
**Pull request by mometwaly**

### CODE REVIEW

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -1,8 +1,19 @@
 import { NgModule } from '@angular/core';
 import { Routes, RouterModule } from '@angular/router';
+import { CategoriesComponent } from './pages/categories/categories.component';
+import { VeterinaryComponent } from './components/veterinary/veterinary.component';
```
Comment (almgwary) [inline src/app/app-routing.module.ts:4]: Please make component name abstract like product or item

File: src/app/app-routing.module.ts [hunk 2]
```diff
@@ -1,8 +1,19 @@
...
+import { CategoryItemComponent } from './pages/category-item/category-item.component';
+import { ShoppingCartComponent } from './components/shopping-cart/shopping-cart.component';
 
 
-const routes: Routes = [];
+const routes: Routes = [
+  { path: '', redirectTo: '/categories', pathMatch: 'full' },
+  { path: 'categories', component: CategoriesComponent },
+  { path: 'veterinary', component: VeterinaryComponent },
```
Comment (almgwary) [inline src/app/app-routing.module.ts:12]: Please make route name abstract like product or item

File: src/app/components/category/category.component.html [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<figure class="card card-product">
+    <a [routerLink]="category | lowercase">
```
Comment (almgwary) [inline src/app/components/category/category.component.html:2]: please don't use pipe here, it is not necessary, you can add new property called link

File: src/app/components/category/category.component.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { Component, OnInit, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-category',
+  templateUrl: './category.component.html',
+  styleUrls: ['./category.component.scss']
+})
+export class CategoryComponent implements OnInit {
+  @Input() category: string;
```
Comment (almgwary) [inline src/app/components/category/category.component.ts:9]: please reuse category and image in new class called category with name, image, link

File: src/app/components/header/header.component.html [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      </li>
+      <li class="nav-item" routerLinkActive="active">
+        <a class="nav-link" [routerLink]="['/categories/products']">Product</a>
+      </li>
+      <li class="nav-item" routerLinkActive="active">
+        <a class="nav-link" [routerLink]="['/veterinary']">Veterinary Problem</a>
+      </li>
+      <li class="nav-item" routerLinkActive="active">
+        <a class="nav-link" [routerLink]="['/shopping-cart']">Shopping Cart</a>
+      </li>
```
Comment (almgwary) [inline src/app/components/header/header.component.html:30]: @mometwaly  please use ngFor with array from ts file

File: src/app/components/items/items.component.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-items',
+  templateUrl: './items.component.html',
+  styleUrls: ['./items.component.scss']
+})
+export class ItemsComponent implements OnInit {
```
Comment (almgwary) [inline src/app/components/items/items.component.ts:8]: why component name is plural and it only represents on item ?

File: src/app/components/shopping-cart/shopping-cart.component.html [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+              <td>Vans Sk8-Hi MTE Shoes</td>
+              <td>89$</td>
+              <td class="qty"><input type="text" class="form-control" id="input1" value="2"></td>
+              <td>178$</td>
+              <td>
+                <a href="#" class="btn btn-danger btn-sm">
+                  <i class="fa fa-times"></i>
+                </a>
+              </td>
+            </tr>
```
Comment (almgwary) [inline src/app/components/shopping-cart/shopping-cart.component.html:45]: @mometwaly  please read row data from an array with ngFor

File: src/app/components/shopping-cart/shopping-cart.component.html [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+          </tbody>
+        </table> 
+        <div class="d-flex justify-content-end">
+          <h5>Total: <span class="price text-success">89$</span></h5>
```
Comment (almgwary) [inline src/app/components/shopping-cart/shopping-cart.component.html:49]: please read the value from variable

File: src/app/components/veterinary/veterinary.component.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-veterinary',
+  templateUrl: './veterinary.component.html',
+  styleUrls: ['./veterinary.component.scss']
+})
+export class VeterinaryComponent implements OnInit {
```
Comment (almgwary) [inline src/app/components/veterinary/veterinary.component.ts:8]: please change component name and path to an abstract name like item or product

File: src/app/pages/categories/categories.component.ts [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+  selector: 'app-categories',
+  templateUrl: './categories.component.html',
+  styleUrls: ['./categories.component.scss']
+})
+export class CategoriesComponent implements OnInit {
+  categories = ['Cats', 'Dogs', 'Birds', 'Products']
+  images = ['https://zjf683hopnivfq5d12xaooxr-wpengine.netdna-ssl.com/wp-content/uploads/2020/02/GettyImages-1199242002-1-1480x833.jpg',
+    'https://www.sciencemag.org/sites/default/files/styles/article_main_image_-_1280w__no_aspect_/public/dogs_1280p_0.jpg?itok=6jQzdNB8',
+    'https://images2.minutemediacdn.com/image/upload/c_crop,h_1188,w_2121,x_0,y_142/f_auto,q_auto,w_1100/v1554733134/shape/mentalfloss/78996-istock-682216682.jpg',
+    'http://imaginepetproducts.ca/img/imagine-products.png'];
```
Comment (almgwary) [inline src/app/pages/categories/categories.component.ts:13]: please create array of objects, [{name, image, link}]

File: src/app/components/shopping-cart/shopping-cart.component.html (hunk 8 above)
Comment (mometwaly) [reply src/app/components/shopping-cart/shopping-cart.component.html:49]: I suggest when we built our data models we can modify it and i had put it in my consideration when i made the Html template

File: src/app/components/items/items.component.ts (hunk 6 above)
Comment (mometwaly) [reply src/app/components/items/items.component.ts:8]: cause it represent items of a category

**Pull request merged**